src/measure_theory/interval_integral.lean

```diff
@@ -1336,22 +1336,44 @@ begin
 end
 
 /-- Fundamental theorem of calculus-2: If `f : ℝ → E` is continuous on `[a, b]` and has a derivative
-  at `f' x` for all `x` in `[a, b)`, and `f'` is continuous on `[a, b]` and measurable, then
-  `∫ y in a..b, f' y` equals `f b - f a`. -/
+  at `f' x` for all `x` in `(a, b)`, and `f'` is continuous on `[a, b]`, then `∫ y in a..b, f' y`
+  equals `f b - f a`. -/
 theorem integral_eq_sub_of_has_deriv_at' (hcont : continuous_on f (interval a b))
-  (hderiv : ∀ x ∈ Ico (min a b) (max a b), has_deriv_at f (f' x) x)
+  (hderiv : ∀ x ∈ Ioo (min a b) (max a b), has_deriv_at f (f' x) x)
   (hcont' : continuous_on f' (interval a b)) :
   ∫ y in a..b, f' y = f b - f a :=
-integral_eq_sub_of_has_deriv_right hcont (λ x hx, (hderiv x hx).has_deriv_within_at) hcont'
+begin
+  refine integral_eq_sub_of_has_deriv_right hcont _ hcont',
+  intros y hy,
+  rw [Ico, mem_set_of_eq, le_iff_lt_or_eq, or_and_distrib_right, ← mem_Ioo] at hy,
+  cases hy,
+  { exact (hderiv y hy).has_deriv_within_at },
+  { refine has_deriv_at_interval_left_endpoint_of_tendsto_deriv
+      (λ x hx, (hderiv x hx).has_deriv_within_at.differentiable_within_at)
+        ((hcont y (Ico_subset_Icc_self (mem_Ico.mpr ⟨hy.1.le, hy.2⟩))).mono Ioo_subset_Icc_self)
+          _ _,
+    { rw [hy.1, ← nhds_within_Ioc_eq_nhds_within_Ioi hy.2,
+          mem_nhds_within_iff_exists_mem_nhds_inter],
+      exact ⟨Ico (y-1) (max a b), Ico_mem_nhds (by linarith) hy.2, by { assume c hc,
+        simpa only [inter_def, Ioc, Ico, mem_set_of_eq] using mem_Ioo.mpr ⟨hc.2.1, hc.1.2⟩ }⟩ },
+    { have hf := (hcont'.continuous_within_at (left_mem_Icc.mpr min_le_max)),
+      rw [interval, hy.1] at hf,
+      have hf' : tendsto f' (𝓝[Ici y] y) (𝓝 (f' y)) :=
+        by convert hf using 1; rw ← nhds_within_Icc_eq_nhds_within_Ici hy.2,
+      have hcongr : deriv f =ᶠ[𝓝[Ioi y] y] f' :=
+        by simpa only [eventually_eq] using eventually_of_mem (Ioo_mem_nhds_within_Ioi
+          (by simpa only [hy.1] using left_mem_Ico.mpr hy.2)) (λ x hx, (hderiv x hx).deriv),
```

Comment:
```suggestion
      have hcongr : deriv f =ᶠ[𝓝[Ioi y] y] f',
      { simpa only [eventually_eq] using eventually_of_mem (Ioo_mem_nhds_within_Ioi
          (by simpa only [hy.1] using left_mem_Ico.mpr hy.2)) (λ x hx, (hderiv x hx).deriv) },
```
Braces are more readable here (rather than `:=`, which is better for a short and/or pure term mode proof).  Also, can you rewrite this one (like the other one, lines 1357-8) to not have the long tactic application `(by simpa only ...)` inside a term?